Sources/Commands/SwiftTestCommand.swift

```diff
@@ -725,6 +726,17 @@ extension SwiftTestCommand {
         var _deprecated_passthrough: Bool = false
 
         func run(_ swiftCommandState: SwiftCommandState) async throws {
+            do {
+                try await self.runCommand(swiftCommandState)
+            } catch let error as FileSystemError {
+                if sharedOptions.shouldSkipBuilding {
+                    throw ErrorWithContext(error, "Test build artifacts were not found in the build folder. Found the --skip-build flag; build the tests first or rerun the command without --skip-build")
```

Comment:
The wrapped error contains the detailed information about what file wasn't found. Here is the error message as it stands right now:

```
error: 	Test build artifacts were not found in the build folder.
	The `--skip-build` flag was provided; either build the tests first with `swift build --build tests` or rerun the `swift test list` command without `--skip-build`
	/Users/plemarquand/work/testapps/DelMeDelMe/.build/arm64-apple-macosx/debug/DelMeDelMePackageTests.xctest doesn't exist in file system
```

Because the xctest codepath throws a FileSystemError as soon as it encounters a missing file we'd need to rework the architecture a bit to keep iterating and collect up all the errors in order to show a list of all the files that aren't found. I can go that route if need be but its going to make the code more complex for a small gain I think.